packages-next/fields-document/src/DocumentEditor/insert-menu.test.tsx

```diff
@@ -0,0 +1,360 @@
+        <text>
+          a
+          <cursor />
+        </text>
+      </paragraph>
+    </editor>
+  `);
+});
+
+test('insertMenu mark is removed when whitespace is inserted', () => {
+  let editor = makeEditor(
+    <editor marks={{ insertMenu: true }}>
+      <paragraph>
+        <text insertMenu>
+          /something
+          <cursor />
+        </text>
+      </paragraph>
+    </editor>,
+    { normalization: 'skip' }
+  );
+
+  editor.insertText(' ');
+
+  expect(editor).toMatchInlineSnapshot(`
+    <editor>
+      <paragraph>
+        <text>
+          /something 
+          <cursor />
+        </text>
+      </paragraph>
+    </editor>
+  `);
+});
+
+test('insertMenu mark is removed when the cursor is moved outside of the text node', () => {
+  let editor = makeEditor(
+    <editor marks={{ insertMenu: true }}>
+      <paragraph>
+        <text insertMenu>
+          /something
+          <cursor />
+        </text>
+      </paragraph>
+      <paragraph>
+        <text />
+      </paragraph>
+    </editor>,
+    { normalization: 'skip' }
+  );
+
+  Transforms.move(editor, { unit: 'character' });
+  expect(editor).toMatchInlineSnapshot(`
+    <editor>
+      <paragraph>
+        <text>
+          /something
+        </text>
+      </paragraph>
+      <paragraph>
+        <text>
+          <cursor />
+        </text>
+      </paragraph>
+    </editor>
+  `);
+});
+
+test('insertMenu mark is added when inserting / at the start of a block', () => {
+  let editor = makeEditor(
+    <editor>
+      <paragraph>
+        <text>
+          <cursor />
+        </text>
+      </paragraph>
+    </editor>
+  );
+
+  editor.insertText('/');
+  expect(editor).toMatchInlineSnapshot(`
+    <editor
+      marks={
+        Object {
+          "insertMenu": true,
+        }
+      }
+    >
+      <paragraph>
+        <text
+          insertMenu={true}
+        >
+          /
+          <cursor />
+        </text>
+      </paragraph>
+    </editor>
+  `);
+});
+
+test('insertMenu mark is added when inserting / after whitespace', () => {
+  let editor = makeEditor(
+    <editor>
+      <paragraph>
+        <text>
+          content <cursor />
+        </text>
+      </paragraph>
+    </editor>
+  );
+
+  editor.insertText('/');
+  expect(editor).toMatchInlineSnapshot(`
+    <editor
+      marks={
+        Object {
+          "insertMenu": true,
+        }
+      }
+    >
+      <paragraph>
+        <text>
+          content 
+        </text>
+        <text
+          insertMenu={true}
+        >
+          /
+          <cursor />
+        </text>
+      </paragraph>
+    </editor>
+  `);
+});
+
+test('insertMenu mark is added when inserting / after whitespace when there is content after the cursor', () => {
+  let editor = makeEditor(
+    <editor>
+      <paragraph>
+        <text>
+          content <cursor /> more
+        </text>
+      </paragraph>
+    </editor>
+  );
+
+  editor.insertText('/');
+  expect(editor).toMatchInlineSnapshot(`
```

Comment:
Maybe add a comment to note the cursor behaviour difference between the browser and snapshot